ethers-providers/src/rpc/provider.rs

```diff
@@ -1443,13 +1444,25 @@ impl ProviderExt for Provider<HttpProvider> {
 /// ```
 /// use ethers_providers::is_local_endpoint;
 /// assert!(is_local_endpoint("http://localhost:8545"));
+/// assert!(is_local_endpoint("http://169.254.0.0:8545"));
 /// assert!(is_local_endpoint("http://127.0.0.1:8545"));
+/// assert!(!is_local_endpoint("http://206.71.50.230:8545"));
+/// assert!(!is_local_endpoint("havenofearlucishere"));
 /// ```
 #[inline]
-pub fn is_local_endpoint(url: &str) -> bool {
-    url.contains("127.0.0.1") || url.contains("localhost")
+pub fn is_local_endpoint(endpoint: &str) -> bool {
+    if let Ok(url) = Url::parse(endpoint) {
+        if let Some(host) = url.host() {
+            match host {
+                Host::Domain("localhost") => return true,
+                Host::Ipv4(ipv4) => return ipv4 == Ipv4Addr::LOCALHOST || ipv4.is_link_local(),
+                // skipping IPV6 check
+                _ => return false,
```

Comment:
Any reason not to include the same for `Ipv6Addr`? (`ipv6.is_loopback()`)

Reply:
good point loopback is available for both variants

Reply:
I did not support Ipv6 addresses since it looked like they were not being used where this function is being called.

If we need I can always add.

@mattsse should I?

Reply:
Added with [supporting also ipv6](https://github.com/gakonst/ethers-rs/pull/2351/commits/cad99e713d1aae97b57db6e9f910286537e0e053)